### PR TITLE
Hard crash in InstrumentViewer while 'extracting the mask'

### LINKED ISSUE #31521: Hard crash in InstrumentViewer while 'extracting the mask'
Mantid crashes with a Segmentation fault when any D2B data available with Mantid is loaded and the InstrumentViewer is started on that workspace. There is a dialog popup which displays the following message: "An error occurred when extracting the mask.". 

I am still investigating this but right now I believe the issue is caused by failing to extract mask workspace from the ADS (L786), but the handling of the exception and the provided message are not helpful. 

### Steps to reproduce the behavior
1. In workbench, load any data from the Testing/Data/SystemTest/ILL/D2B 
2. Show instrument on the loaded workspace.
3. Observe the warning message and subsequent crash.

### Platforms affected
workbench

### PR DESCRIPTION
**Pull request by darominski**
This PR fixes the hard crash found when trying to interact with InstrumentViewer on a `ScanningWorkspace`, that does is not supported. This issue is  caused by non-existing mask workspace. This small fix leads to expected flow resulting in a warning message for the user, explaining (in a way) that this workspace type is not supported: 
```
Could not show instrument for workspace 'workspace_id':
DetectorInfo accessed without time index but the beamline has time-dependent (moving) detectors.
```

While an exception related to this mask workspace being a `nullptr` is caught in L802 of InstrumentActor.cpp, the issue then propagates until there was an attempt to call a `isMasked` method on a `nullptr` workspace resulting in a segmentation violation. A warning dialog is displayed to the user with a message that might not be something they expect. I propose to add a line to this message that attempts to explain the situation more: 'This workspace may not be supported by the InstrumentView', but please provide your suggestions.

**To test:**

1. Load any D2B data from either UnitTest or SystemTest directory, e.g. SystemTest/ILL/D2B/532009.nxs
2. Show instrument.

There should be no hard crash and a warning dialog should be displayed and after that the lines shown above describing that this workspace cannot be displayed.

Fixes #31521 . 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/widgets/instrumentview/src/InstrumentActor.cpp [hunk 1]
```diff
@@ -1058,8 +1059,8 @@ void InstrumentActor::setDataIntegrationRange(const double &xmin,
       // Ignore monitors if they are masked on the view
       if (spectrumDefinition.size() == 1 &&
           (std::find(monitorIndices.begin(), monitorIndices.end(), i) !=
-               monitorIndices.end() ||
-           maskWksp->isMasked(static_cast<int>(i))))
+               monitorIndices.end() || (maskWksp != nullptr &&
```
Comment (DavidFair) [inline qt/widgets/instrumentview/src/InstrumentActor.cpp:1062]: Works for me, the only nit tweaking this to get rid of the comparison then it matches the common usage across the codebase:

`(maskWksp && maskWksp->isMasked(...))`

**Review by DavidFair — CHANGES_REQUESTED** [review]
I'm happy to approve this from a functional POV

I'm not in tomorrow (Friday), but I've tested it with the suggested line too. So from a gatekeeper POV feel free to assume I've approved this with said suggestion too, so we can get this quickly merged

File: qt/widgets/instrumentview/src/InstrumentActor.cpp (hunk 1 above)
Comment (darominski) [reply qt/widgets/instrumentview/src/InstrumentActor.cpp]: Thank you for having a look into this. I removed the comparison and also modified the warning message when the ExtractMask algorithm fails to include information that scanning workspaces are not supported yet, as suggested by @gvardany .

**Review by DavidFair — DISMISSED** [review]

**Comment by gvardany** [comment]
Please rebase to release-next.

**Review by DavidFair — APPROVED** [review]

**Pull request merged**